### PR TITLE
Added option to generate the api reference in docbook format and prov…

### PR DESCRIPTION
**Pull request by kc284**
…ided relevant template.

Signed-off-by: Konstantina Chremmou <konstantina.chremmou@citrix.com>

### CODE REVIEW

File: ocaml/doc/doc-convert.sh [hunk 1]
```diff
@@ -7,22 +7,48 @@ GRAPHVIZ_FLAGS="-Tpng"
 dot vm-lifecycle.dot ${GRAPHVIZ_FLAGS} -o vm-lifecycle.png
 dot classes.dot ${GRAPHVIZ_FLAGS} -o classes.png
 
-if  [ ! "$1" = "--pdf" ]; then
+usage() {
+    echo "Usage:"
+    echo "    sh $0 [--pdf|--db]"
```
Comment (mseri) [inline ocaml/doc/doc-convert.sh]: `--docbook` instead of `--db`?

**Comment by mseri** [comment]
I'd start considering converting `docconvert.sh` to `python` and use `argcomplete` to specify the options if we think we could add more features sooner or later. It starts getting too long for a shell script.

File: ocaml/doc/doc-convert.sh (hunk 1 above)
Comment (kc284) [inline ocaml/doc/doc-convert.sh]: Isn't db intuitive enough? 😝 😄 
👍

**Comment by kc284** [comment]
Is python installed by default on recent linux distros (or at least on our build containers)?

**Comment by mseri** [comment]
Even in very non recent ones

**Comment by kc284** [comment]
@mseri as discussed, I changed the switch so we can merge this and I'll make at a later time a different PR to convert to python, because it requires further changes in the sdk build.

**Pull request merged**